mmv1/templates/terraform/decoders/memorystore_instance.go.tmpl

```diff
@@ -85,10 +85,31 @@
 					}
 				}
 			}
+			// NOTE: Due to how the api updates the backend. When mode is set to CLUSTER_DISABLED
+			// When desired_auto_created_endpoints is set this flow works as expected
+			// FLOW: Terraform  -> API Input -> API Output -> Terraform output
+			// 1. CLUSTER_DISABLED and desired_auto_created_endpoints defined: desired_auto_created_endpoints -> endpoints -> Endpoints -> desired_auto_created_endpoints + Endpoints
+			// 2. CLUSTER_DISABLED and desired_psc_auto_connections defined: desired_psc_auto_connections -> psc_auto_connections -> Endpoints -> desired_auto_created_endpoints + Endpoints
+			// 3. CLUSTER and desired_psc_auto_connections defined: desired_psc_auto_connections -> psc_auto_connections -> psc_auto_connections -> desired_psc_auto_connections + psc_auto_connection
+			// 4. CLUSTER and desired_auto_created_endpoints defined: desired_auto_created_endpoints -> endpoints -> Endpoints -> desired_auto_created_endpoints + Endpoints
+			// The issue is that when CLUSTE_DISABLED is set for that flow  API is updated Endpoints instead of psc_auto_connections
+			// The if else if below checks what is define by the user in order to set the correct fielt in the state
+			// The code below fixes use case 2
+			// The else conditon is to handle importing as there is no clear way to tell what the user has set desired_auto_created_endpoints or desired_psc_auto_connections
+			// Default is to set endpoints and user will need to adjust thier Terraform config as to use desired_auto_created_endpoints
```

Comment:
This comment is currently describing what's being fixed in this PR. Could you instead make it explain why the code is written the way it is? Something like: "we want to make these fields detect API-side drift, so if the API returns a value for them we set them in state."